src/librbd/cache/pwl/ssd/WriteLog.cc

```diff
@@ -350,10 +350,6 @@ void WriteLog<I>::setup_schedule_append(pwl::GenericLogOperationsVector &ops,
                                         bool do_early_flush,
                                         C_BlockIORequestT *req) {
   this->schedule_append(ops);
```

Comment:
@hualongfeng did you mean `req` is getting deleted in this expression? Could you please point to the code where you think `req` is getting deleted?

Also can you please confirm that the ssd unit tests pass with this PR?

Reply:
> @hualongfeng did you mean `req` is getting deleted in this expression? Could you please point to the code where you think `req` is getting deleted?
> 
> Also can you please confirm that the ssd unit tests pass with this PR?
> 
> Edit: I tested this - `TestMockCacheSSDWriteLog.flush_source_user` won't pass since the request is not completed
@MahatiC  Now, I don't know where req is getting deleted.  But after line 352, req->m_finish_called will be true, this means that the req has been deleted.

Reply:
@MahatiC   how to do ssd unit test?

Reply:
@MahatiC  . I thinks https://github.com/ceph/ceph/blob/d82a565f0e4371ea3a47c9d643f45c52311422ba/src/librbd/cache/pwl/Request.cc#L88 can call : complete_user_request and release_cell. So why you call this in func setup_schedule_append?

Reply:
I trace the code with gdb, I found the finished ctx is appended to m_work_queue. Can it be understood here that the task performed by the current thread and the thread where the work queue finished ctx is located will execute first? If the current thread is before the finish, the program runs normally, otherwise it will cause illegal access. @MahatiC

Reply:
> @MahatiC how to do ssd unit test?

```
cd build
ninja
./bin/unittest_librbd --gtest_filter=TestMockCacheSSDWriteLog.flush_source_user
```

Reply:
> can call : complete_user_request and release_cell. So why you call this in func setup_schedule_append?

@majianpeng to facilitate different actions for `rwl` and `ssd. See:

https://github.com/ceph/ceph/blob/d82a565f0e4371ea3a47c9d643f45c52311422ba/src/librbd/cache/pwl/Request.cc#L81

Reply:
> > @MahatiC how to do ssd unit test?
> 
> @majianpeng
> 
> ```
> cd build
> ninja
> ./bin/unittest_librbd --gtest_filter=TestMockCacheSSDWriteLog.flush_source_user
> ```

Why github pass all unittest? or github qa don't do unittst_librbd?

Reply:
@majianpeng Github CI does run unitests, it just doesn't run the client-side cache tests as they're not enabled[0]

[0]https://github.com/ceph/ceph/blob/master/src/test/librbd/CMakeLists.txt#L135